src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -60,15 +68,71 @@ export class CourseExercisesComponent implements OnInit, OnDestroy {
         this.paramSubscription.unsubscribe();
     }
 
-    public groupExercises(selectedOrder: number): void {
+    /**
+     * Return the total number of exercises for the logged in user
+     */
+    getNrOfExercises(): number {
+        return [...this.exerciseCountMap.values()].reduce((a, b) => a + b, 0);
+    }
+
+    /**
+     * Reorders all displayed exercises
+     * @param selectedOrder The order in which the exercises should be displayed
+     */
+    orderUpdate(selectedOrder: number) {
+        this.order = selectedOrder;
+        this.applyFiltersAndOrder();
+    }
+
+    /**
+     * Filters all displayed exercises by applying the selected filters
+     * @param filters The filters which should be applied
+     * @param active Should the filters be active or inactive
+     */
+    filterUpdate(filters: number[], active: boolean) {
```

Comment:
Opinionated:
I think it would be better to not use an active parameter, but to toggle the filters from the activeFilters set.
I would also propose to bind the call `activeFilters.includes(FILTER)` to the input in the template:
`[checked]="activeFilters.includes(FILTER)`.
This way you can be sure that the component and the template are always in sync.

Reply:
Checked is now bound to the template